lib/hammer_cli_foreman/host.rb

```diff
@@ -305,6 +315,13 @@ def validate_options
         end
       end
 
+      def request_params
+        params = super
+        if option_image_id
+          HammerCLIForeman::Host.set_image_uuid(params,option_image_id)
```

Comment:
If we split (Single Responsibility) the `set_image_uuid` we could move getting the image to ComputeResource which would be more expected place for the code. I also don't think HammerCLIForeman::Host super-command is a good place  for storing helper methods for individual subcommands. I'd keep the assignment in the Update/Create commands. If you'd need to make it more DRY it would be better to create a module with the helper method that is included in the subcommands. 
```ruby
compute_resource_id = params['host']['compute_resource_id']
image_uuid = HammerCLIForeman::ComputeResource::Utils.image_uuid(compute_resource_id, options['option_image_id'])
params['host']['compute_attributes']['image_id'] = image uuid 
```

Reply:
we could move getting the image to ComputeResource  - which file?